pkgs/desktops/gnome/misc/gnome-flashback/default.nix

```diff
@@ -155,15 +156,38 @@ let
             '';
           };
 
-          executable = writeShellScriptBin "gnome-flashback-${wmName}" ''
-            if [ -z $XDG_CURRENT_DESKTOP ]; then
-              export XDG_CURRENT_DESKTOP="GNOME-Flashback:GNOME"
-            fi
-
-            export XDG_DATA_DIRS=${wmApplication}/share:${gnomeSession}/share:${gnome-flashback}/share:${gnome-panel}/share:$XDG_DATA_DIRS
+          # gnome-panel will only look for applets in a single directory so symlink them into here.
+          panelModulesEnv = buildEnv {
+            name = "panel-modules-env";
+            # We always want to find the built-in panel applets.
+            paths = [ gnome-panel gnome-flashback ] ++ panelModulePackages;
+            pathsToLink = [ "/lib/gnome-panel/modules" ];
+          };
 
-            exec ${gnome-session}/bin/gnome-session --session=gnome-flashback-${wmName} "$@"
-          '';
+          executable = stdenv.mkDerivation {
+            name = "gnome-flashback-${wmName}";
+            nativeBuildInputs = [ wrapGAppsHook ];
+            buildInputs = [ glib gnome-panel gnome-flashback ] ++ panelModulePackages;
```

Comment:
The panel is still here and in XDG_DATA_DIR.

Reply:
It needs to be in XDG_DATA_DIR so GNOME can find its .desktop file.

Reply:
Yeah, but is that needed even with `enableGnomePanel`?

Also the reason that `XDG_DATA_DIRS` are for desktop files might be nice to mention in a comment next to it.

Reply:
Or is gnome-panel required by something even with `enableGnomePanel = false`?

Reply:
Done.